vassal-app/src/main/java/VASSAL/counters/Decorator.java

```diff
@@ -73,15 +75,29 @@ public abstract class Decorator extends AbstractImageFinder implements EditableP
   PropertyExporter, SearchTarget, ImageSearchTarget {
 
   protected GamePiece piece;
+  private GamePiece innermost;
   private Decorator dec;
   private boolean selected = false;
 
+  private static Set<String> innerMostProperties = new HashSet<>();
+  static {
+    innerMostProperties.add(BasicPiece.LOCATION_NAME);
+    innerMostProperties.add(BasicPiece.DECK_NAME);
+    innerMostProperties.add(BasicPiece.CURRENT_BOARD);
+    innerMostProperties.add(BasicPiece.CURRENT_MAP);
+    innerMostProperties.add(BasicPiece.CURRENT_ZONE);
+    innerMostProperties.add(BasicPiece.CURRENT_X);
+    innerMostProperties.add(BasicPiece.CURRENT_Y);
+  }
```

Comment:
What happens if some trait duplicates one of these?

Reply:
That's a risk, but feeling is a) unlikely and b) tough.

Reply:
hohoho

Reply:
Happy to remove if we feel it is too much. I restrained myself from adding all the Old... properties as well.

Reply:
Seems worth keeping to me.

Reply:
Have we verified that this makes much of a difference?

Reply:
[This Discord reply](https://discord.com/channels/740127632586178572/740127633215586337/1101346924243722250) to @BrentEaston reports a 30%+ improvement in chain of ranged GKCs.

I've don't personally have such an obvious case to test. I haven't seen any apparent detriment and the associated functions are more targeted, which simplifies the search expressions that they require.

Reply:
I believe the 30% improvement is primarily due to the range checking improvements, I doubt adding these CURRENT properties to the innermost handling does much at all. I am happy to remove these.

Reply:
I'm worried that these are going to result in hard-to-troubleshoot problems; I think we ought not to do this.

Reply:
I made one more minor comment elsewhere. Otherwise, I think this PR is ok now.

Reply:
Removed